aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,246 @@
+
+```csharp
+[JSImport("getMessage", "Contoso.InteropServices.JavaScript.Interop")]
+```
+
+If the JS function doesn't directly interact with the rendered Document Object Model (DOM), import the module in [`OnInitializedAsync`](xref:blazor/components/lifecycle#component-initialization-oninitializedasync). Call the imported JS function with the .NET interop method.
+
+In the following `CallJavaScript` component:
+
+* The `Interop` module is imported asychronously from the [collocated JS file](xref:blazor/js-interop/index#load-a-script-from-an-external-javascript-file-js-collocated-with-a-component).
+* The imported `getMessage` JS function is called with `Interop.GetWelcomeMessage()`.
+* The returned welcome string is displayed in the UI via the `message` field.
+
+`Pages/CallJavaScript.razor`:
+
+```razor
+@page "/call-javascript"
+@using System.Runtime.InteropServices.JavaScript
+
+<h1>.NET JS <code>[JSImport]</code>/<code>[JSExport]</code> interop (Call JS)</h1>
+
+@(message is not null ? message : string.Empty)
+
+@code {
+    private string? message;
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (OperatingSystem.IsBrowser())
+        {
+            await JSHost.ImportAsync("Interop", "../Pages/CallJavaScript.razor.js");
+
+            message = Interop.GetWelcomeMessage();
+        }
+    }
+}
+```
+
+> [!NOTE]
+> The conditional check for <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType> in the preceding example ensures that the code is only called in Blazor WebAssembly apps running on the client. This is important for library code deployed as NuGet packages that might be referenced by developers of Blazor Server apps, where the preceding code can't execute. If you know for sure that the code is only implemented in Blazor WebAssembly apps, you can remove the check for <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+If the JS module interacts directly with the component's rendered UI, import the module in [`OnAfterRenderAsync`](xref:blazor/components/lifecycle#after-component-render-onafterrenderasync) and call the imported JS function:
+
+```csharp
+protected override async Task OnAfterRenderAsync(bool firstRender)
+{
+    if (OperatingSystem.IsBrowser() && firstRender)
+    {
+        await JSHost.ImportAsync("Interop", "../Pages/CallJavaScript.razor.js");
+
+        Interop.DoSomething();
+    }
+}
+```
+
+If the code holds a reference to the <xref:System.Runtime.InteropServices.JavaScript.JSObject> returned by `JSHost.ImportAsync`, be sure to call <xref:System.Runtime.InteropServices.JavaScript.JSObject.Dispose%2A?displayProperty=nameWithType> when the component is disposed:
```

Comment:
This is interesting comment. The `JSObject` returned by the `ImportAsync` is indeed disposable proxy. But even if you dispose it, the `JSImport` infrastructure will keep the JS side ES6 module alive and callable.